var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -87,14 +87,22 @@ class Trilinos(CMakePackage):
             description='Enables the build of shared libraries')
     variant('debug',        default=False,
             description='Builds a debug version of the libraries')
-    variant('boost',        default=True, description='Compile with Boost')
-    variant('tpetra',       default=True, description='Compile with Tpetra')
-    variant('exodus', default=False, description='Compile with Exodus from SEACAS')
+    variant('boost',        default=True,  description='Compile with Boost')
+    variant('tpetra',       default=True,  description='Compile with Tpetra')
+    variant('exodus',       default=False, description='Compile with Exodus from SEACAS')
+
+    variant('dtk',          default=False, description='Enable DataTransferKit')
+    resource(name='dtk',
+             git='https://github.com/ornl-cees/DataTransferKit',
```

Comment:
> Does it mean if I provide the hash it's going to behave like a release?

not sure about that. This is often done in git-only packages without stable releases to pretend that there is at least one stable release `0.0.1`. 

> How do I do that?

i could not find docu, but one can specify `tag=` (see http://spack.readthedocs.io/en/latest/packaging_guide.html?highlight=resource#resources-expanding-extra-tarballs), maybe `hash=` works as well. If not, don't bother.

>  is it possible to provide multiple versions for a resource?

for sure. Look at `llvm` or `fenics` for examples. One hacky way could be making your variant multi-valued, i.e. `dtk=master` or `dtk=0.0.1` and then use it in `when` of resource. This might work, but I am not sure if it worths the hassles.